src/Modules/ExteriorPowers/FreeModules.jl

```diff
@@ -47,9 +47,9 @@ function exterior_power(F::FreeMod, p::Int; cached::Bool=true)
     @req parent(u) === result "element does not belong to the correct module"
     k = findfirst(==(u), gens(result))
     @req !isnothing(k) "element must be a generator of the module"
-    ind = ordered_multi_index(k, p, n)
+    ind = combination(n, p, k)
     e = gens(F)
-    return Tuple(e[i] for i in indices(ind))
+    return Tuple(e[i] for i in ind)
```

Comment:
```suggestion
    return Tuple(gen(F, i) for i in ind)
```
this only accesses a subset of the gens, so it is better not not allocate all of them but only the ones we actually need.